### ISSUE TITLE
[FEATURE] Replace go-multierror with standard errors library

### ISSUE THREAD

**Issue by Kern--** [awslabs/soci-snapshotter#793]
### Description

[Go 1.20 introduced multiple error wrapping](https://tip.golang.org/doc/go1.20#errors) which allows errors to be combined e.g.:
```
err := fmt.Errorf("context: %w: %w", err1, err2)
err := errors.Join(err1, err2).
```

We have been using [go-multierror](https://github.com/hashicorp/go-multierror) to do this same sort of wrapping like
```
var err error
err = multierror.Append(err, err1)
err = multierror.Append(err, err2)
```

go-multierror has it's own API for introspecting the wrapped errors and formatting the response, but we never used any of them. That makes our use of go-multierror and the new error wrapping nearly identical except that calling `err.Error()` on a [go-multierror only returns the first error's message.](https://github.com/hashicorp/go-multierror/blob/main/multierror.go#L100) If we switch, we would get the full set of errors.

### Describe the solution you'd like

Switch to the standard errors library and remove go-multierror from our direct dependencies in both `go.mod` and `cmd/go.mod`

### Any additional context or information about the feature request

PR discussion that brought this up https://github.com/awslabs/soci-snapshotter/pull/792#discussion_r1298785109

**Comment by manujgrover71** [comment]
@Kern-- I would to work on this.

**Comment by manujgrover71** [comment]
>  Switch to the standard errors library and remove go-multierror from our direct dependencies in both go.mod and cmd/go.mod

go-multierror is in indirect dependancy only in cmd/go.mod

**Comment by turan18** [comment]
Resolved in #795
